cashier-google-play-billing/src/main/java/com/getkeepsafe/cashier/billing/GooglePlayBillingApi.java

```diff
@@ -0,0 +1,267 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.getkeepsafe.cashier.billing;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.UiThread;
+import android.text.TextUtils;
+import com.android.billingclient.api.BillingClient;
+import com.android.billingclient.api.BillingClient.BillingResponse;
+import com.android.billingclient.api.BillingClient.FeatureType;
+import com.android.billingclient.api.BillingClient.SkuType;
+import com.android.billingclient.api.BillingClientStateListener;
+import com.android.billingclient.api.BillingFlowParams;
+import com.android.billingclient.api.ConsumeResponseListener;
+import com.android.billingclient.api.Purchase;
+import com.android.billingclient.api.SkuDetails;
+import com.android.billingclient.api.SkuDetailsParams;
+import com.android.billingclient.api.SkuDetailsResponseListener;
+import com.getkeepsafe.cashier.logging.Logger;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public final class GooglePlayBillingApi extends AbstractGooglePlayBillingApi implements BillingClientStateListener {
+
+    /** Internal log tag **/
+    private static final String LOG_TAG = "GoogleBillingApi";
+
+    /** Google Play Billing client **/
+    private BillingClient billing;
+
+    /** Google Play Billing service life cycle listener **/
+    private LifecycleListener listener;
+
+    /** Google Play Billing connection state **/
+    private boolean isServiceConnected = false;
+
+    private Handler mainHandler = new Handler(Looper.getMainLooper());
+
+    @Override
+    public boolean initialize(final @NonNull Context context, final @NonNull GooglePlayBillingVendor vendor,
+                              LifecycleListener listener, Logger logger) {
+        final boolean initialized = super.initialize(context, vendor, listener, logger);
+        this.listener = listener;
+
+        if (available() && listener != null) {
+            listener.initialized(true);
+            return true;
+        }
+
+        // Google Billing require client creation to be performed on main thread.
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            createClient(context, vendor);
+        } else {
+            mainHandler.post(new Runnable() {
+                @Override
+                public void run() {
+                    createClient(context, vendor);
+                }
+            });
+        }
+
+        return initialized;
+    }
+
+    @UiThread
+    private void createClient(@NonNull Context context, @NonNull GooglePlayBillingVendor vendor) {
```

Comment:
I suggest this returns a `Runnable` and renamed as `createClientTask` such as:

```
private Runnable createClientTask(@NonNull final Context context, @NonNull final GooglePlayBillingVendor vendor) {
        return new Runnable() {
            @Override
            public void run() {
                logSafely("Creating Google Play Billing client...");
                billing = BillingClient.newBuilder(context)
                        .setListener(vendor)
                        .build();

                logSafely("Attempting to connect to billing service...");
                billing.startConnection(GooglePlayBillingApi.this);
            }
        };
    }
```

This would make `#initialize` easier to read.